File: tests/test_scala/com/example/services/UserService.scala
package com.example.services

import com.example.models.User
import scala.io.Source

class UserService {
    def save(user: User): Unit = {
        println(s"Saving user: ${user.name}")
    }

    def findAll: List[User] = {
        List.empty[User]
    }

    def update(user: User): Boolean = {
        println(s"Updating user: ${user.name}")
        true
    }
}

File: tests/test_kotlin/com/example/services/UserService.kt
package com.example.services

import com.example.models.User
import java.io.PrintStream

class UserService {
    private val logger = PrintStream(System.out)

    fun save(user: User) {
        logger.println("Saving user: ${user.name}")
    }

    fun findAll(): List<User> {
        return emptyList()
    }

    fun update(user: User): Boolean {
        logger.println("Updating user: ${user.name}")
        return true
    }
}
